### DISCUSSION TITLE
Does Railway actually have ~241 GB RAM?

### DISCUSSION THREAD

**Discussion by AyaanZaveri** [coder/code-server#4019] [General] [answered]
Neofetch Results:

![Railway](https://user-images.githubusercontent.com/63752541/130525827-040fc647-cc8f-4681-b12e-57150b619008.jpeg)

Also, is it possible to change the distro from Debian to Ubuntu?

**Comment by jsjoeio** [comment]
Oh wow I have no idea 👀

I think that might be a question for the Railway team. 

> Also, is it possible to change the distro from Debian to Ubuntu?

🤔 I'm not sure if this is possible. Maybe @bpmct knows

**Comment by AyaanZaveri** [comment]
Ok, Thanks!

**Comment by bpmct** [comment, marked as answer]
You do not have access to all of those resources in the environment. I believe that is just a sideeffect of how Railway manages their containers. Here are the resources your environment would have: https://railway.app/pricing

> Also, is it possible to change the distro from Debian to Ubuntu?

It is, the code-server Dockerfile uses Debian, but you could make a custom image by combining the [code-server Dockerfile](https://github.com/cdr/code-server/blob/main/ci/release-image/Dockerfile) and the [deploy-code-server one you have in your repo](https://github.com/AyaanZaveri/deploy-code-server/blob/main/Dockerfile).

Here is an example (change the Dockerfile in your repo to this, and you should be good to go). Note, this Dockerfile is not well-optimized, but it does use Ubuntu as the base OS, so perhaps you could modify it to suit your needs better 😆 

https://github.com/bpmct/dcs-ubuntu-tests/blob/main/Dockerfile

Some areas for improvement:
- download DEB from releases page 
- remove any stuff not needed for Ubuntu (maybe the uid fix stuff?)
- single entrypoint script instead of downloading one from the code-server repo
- consolidate apt-get installs or separate by "OS" dependencies" and "dev dependencies" for faster builds. (Railway caches all Dockerfile steps before to the first one that was modified)

Let me know if you end up doing this, would be a great doc!

**Reply by nooobcoder** [reply]
@bpmct nice explanation.
